File: kotlinbank/src/main/java/com/m17/damianogiusti/vg2018/actions/ActionFactory.java
package com.m17.damianogiusti.vg2018.actions;

import com.m17.damianogiusti.vg2018.BankAccount;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.io.PrintWriter;
import java.util.Scanner;

public class ActionFactory {

  public static final String DEPOSIT = "1";
  public static final String WITHDRAW = "2";
  public static final String QUIT = "q";

  public UserAction get(@Nullable String command, @NotNull BankAccount bankAccount, PrintWriter writer, Scanner scanner) {
    if (command == null) return null;
    switch (command.toLowerCase()) {
      case ActionFactory.DEPOSIT:
        return new DepositAction(bankAccount, writer, scanner);
      case WITHDRAW:
        return new WithdrawAction(bankAccount, writer, scanner);
      case QUIT:
        return new QuitAction();
      default:
        return null;
    }
  }
}

File: kotlinbank/src/main/kotlin/com/m17/damianogiusti/vg2018/kotlin/actions/ActionFactory.kt
package com.m17.damianogiusti.vg2018.kotlin.actions

import com.m17.damianogiusti.vg2018.kotlin.BankAccount

class ActionFactory {

    operator fun invoke(command: String?, bankAccount: BankAccount): UserAction? {
        return when (command?.toLowerCase()) {
            DEPOSIT -> DepositAction(bankAccount)
            WITHDRAW -> WithdrawAction(bankAccount)
            QUIT -> QuitAction()
            else -> null
        }
    }

    companion object {
        const val DEPOSIT = "1"
        const val WITHDRAW = "2"
        const val QUIT = "q"
    }
}
